eo-maven-plugin/src/main/java/org/eolang/maven/VerifyMojo.java

```diff
@@ -60,11 +65,51 @@ public final class VerifyMojo extends SafeMojo {
 
     @Override
     void exec() throws IOException {
-        throw new UnsupportedOperationException(
-            String.format(
-                "The VerifyMojo is not implemented yet, failOnWarning is %s",
-                this.failOnWarning
+        final Collection<ForeignTojo> tojos = this.scopedTojos().withXmir();
+        final int total = new OptimizedTojos(
+            new Filtered<>(
+                ForeignTojo::notVerified,
+                tojos
+            ),
+            this.optimization(),
+            new OptimizationTask(
+                new MapOf<String, Path>(
+                    new MapEntry<>(OptimizationFolder.TARGET.key(), this.targetDir.toPath()),
+                    new MapEntry<>(OptimizationFolder.CACHE.key(), this.cache)
+                ),
+                new MapOf<String, String>(
+                    new MapEntry<>(OptimizationFolder.TARGET.key(), "6-verify"),
+                    new MapEntry<>(OptimizationFolder.CACHE.key(), "verified")
+                ),
+                ForeignTojo::withVerified,
+                ForeignTojo::shaken
             )
+        ).count();
+        if (total > 0) {
+            Logger.info(
+                this,
+                "Verified %d out of %d XMIR program(s)", total,
+                tojos.size()
```

Comment:
@levBagryansky You can get `int all = tojos.size()` once and in this case you won't need `tojos` variable.

Reply:
@volodya-lombrozo `tojos` is used previously too